discovery-infra/tests/test_validations.py

```diff
@@ -0,0 +1,76 @@
+        self.assert_cluster_validation(orig_cluster, "network", "ingress-vip-defined", "pending")
+        self.assert_cluster_validation(orig_cluster, "network", "ingress-vip-valid", "pending")
+        self.assert_cluster_validation(orig_cluster, "network", "dns-domain-defined", "success")
+        self.assert_cluster_validation(orig_cluster, "network", "cluster-cidr-defined", "success")
+        self.assert_cluster_validation(orig_cluster, "network", "service-cidr-defined", "success")
+        self.assert_cluster_validation(orig_cluster, "network", "no-cidrs-overlapping", "pending")
+        self.assert_cluster_validation(orig_cluster, "network", "network-prefix-valid", "success")
+        self.assert_cluster_validation(orig_cluster, "network", "ntp-server-configured", "success")
+
+        # Check base DNS domain
+        api_client.update_cluster(cluster_id, {"base_dns_domain": ""})
+        self.wait_for_cluster_validation(api_client, cluster_id, "network", "dns-domain-defined", ["failure"])
+        api_client.update_cluster(cluster_id, {"base_dns_domain": orig_cluster.base_dns_domain})
+        self.wait_for_cluster_validation(api_client, cluster_id, "network", "dns-domain-defined", ["success"])
+
+        # Check pull secret
+        api_client.update_cluster(cluster_id, {"pull_secret": ""})
+        self.wait_for_cluster_validation(api_client, cluster_id, "configuration", "pull-secret-set", ["failure"])
+        api_client.update_cluster(cluster_id, {"pull_secret": env_variables['pull_secret']})
+        self.wait_for_cluster_validation(api_client, cluster_id, "configuration", "pull-secret-set", ["success"])
+
+    def test_host_insufficient(self, api_client, cluster, nodes):
+        cluster_id = cluster().id
+        self.generate_and_download_image(cluster_id=cluster_id, api_client=api_client)
+
+        # Modify vCPU count on the first node to be insufficient
+        vcpu_node = nodes.nodes[0]
+        orig_vcpu_value = vcpu_node.get_cpu_cores()
+        vcpu_node.set_cpu_cores(1)
+        vcpu_host_id = vcpu_node.get_host_id()
+
+        # Modify RAM amount on the second node to be insufficient
+        ram_node = nodes.nodes[1]
+        ram_node.set_ram_kib(2097152)  # 2GB
+        ram_host_id = ram_node.get_host_id()
+
+        nodes.start_all()
+        self.wait_until_hosts_are_discovered(cluster_id=cluster_id, api_client=api_client, additional_statuses=[consts.NodesStatus.INSUFFICIENT])
+        self.wait_for_host_validation(api_client, cluster_id, vcpu_host_id, "hardware", "has-min-cpu-cores", ["failure"])
+        self.wait_for_host_validation(api_client, cluster_id, ram_host_id, "hardware", "has-min-memory", ["failure"])
+
+        # Return to original settings and verify that validations pass
+        vcpu_node.shutdown()
+        vcpu_node.set_cpu_cores(orig_vcpu_value)
+        vcpu_node.start()
+
+        ram_node.shutdown()
+        ram_node.set_ram_kib(int(env_variables['master_memory']) * 1024)
```

Comment:
Im thinking now that if test will fail somewhere, then there wont be a clean up for the CPU \ RAM changes (to bring it back to correct values). 
can you please add a fixture for applying a proper cleanup for this test?

Reply:
We reuse VMs between tests?

Reply:
yes, but; 
- stop all nodes
- format disk
- set correct boot order
before\after each test

Reply:
see https://github.com/openshift/assisted-test-infra/blob/master/discovery-infra/tests/base_test.py#L25

Reply:
Done